packages/grid/x-grid/src/tests/filtering.XGrid.test.tsx

```diff
@@ -224,12 +224,13 @@ describe('<XGrid /> - Filter', () => {
       />,
     );
     // TODO should equal 0, the state doesn't change
-    expect(onFilterModelChange.callCount).to.equal(4);
+    expect(onFilterModelChange.callCount).to.equal(8);
```

Comment:
Wait what. It's going backward. I have refactored it in #1760 to be 1, still not 0 but much better than 4.